types/module/module.go

```diff
@@ -627,9 +627,14 @@ func (m *Manager) BeginBlock(ctx sdk.Context, req abci.RequestBeginBlock) abci.R
 	ctx = ctx.WithEventManager(sdk.NewEventManager())
 
 	for _, moduleName := range m.OrderBeginBlockers {
-		module, ok := m.Modules[moduleName].(BeginBlockAppModule)
-		if ok {
+		if module, ok := m.Modules[moduleName].(BeginBlockAppModule); ok {
 			module.BeginBlock(ctx, req)
+		} else if module, ok := m.Modules[moduleName].(appmodule.HasBeginBlocker); ok {
+			err := module.BeginBlock(ctx)
+			if err != nil {
+				// TODO: Handle error
```

Comment:
Please reference the issues for all TODOs :)
```suggestion
				// TODO: Handle error
				//
				// Ref: ...
```